From randy.dunlap@oracle.com  Wed Mar 21 16:01:13 2007
Date: Wed, 21 Mar 2007 13:57:51 -0700
From: Randy Dunlap <randy.dunlap@oracle.com>
Cc: gregkh <greg@kroah.com>
Subject: USB: gadget-storage needs BLOCK
Message-Id: <20070321135751.7ac9886c.randy.dunlap@oracle.com>

From: Randy Dunlap <randy.dunlap@oracle.com>

With CONFIG_BLOCK=n, this build error happens:
WARNING: "bdev_read_only" [drivers/usb/gadget/g_file_storage.ko] undefined!

Signed-off-by: Randy Dunlap <randy.dunlap@oracle.com>
Acked-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/Kconfig |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/gadget/Kconfig
+++ b/drivers/usb/gadget/Kconfig
@@ -370,6 +370,7 @@ config USB_GADGETFS
 
 config USB_FILE_STORAGE
 	tristate "File-backed Storage Gadget"
+	depends on BLOCK
 	help
 	  The File-backed Storage Gadget acts as a USB Mass Storage
 	  disk drive.  As its storage repository it can use a regular
